src/server/models/Error.ts

```diff
@@ -0,0 +1,38 @@
+import { GenericErrors } from '@/shared/model/Errors';
+
+export class ApiError extends Error {
+  status: number;
+  field?: string; // used if the error is related to a particular form field
+
+  constructor({
+    message = GenericErrors.DEFAULT,
+    name = 'ApiError',
+    status = 500,
+    field,
+  }: {
+    message?: string;
+    name?: string;
+    status?: number;
+    field?: string;
+  } = {}) {
+    super();
+    this.message = message;
```

Comment:
will this be identical to `super(message);` ?

Reply:
ah yes! good point, let me refactor slightly